src/coiio/imageio.h

```diff
@@ -0,0 +1,337 @@
+#pragma once
+
+#include "imageio_defines.h"
+#include "typedesc.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// ImageSpec describes the data format of an image -- dimensions, layout,
+/// number and meanings of image channels.
+///
```

Comment:
I'm really not sure yet how we should end up documenting all of this.

But I would advise against copying the whole comment blocks from the C++ side. At the very least, that's going to be a nightmare to keep synchronized.

For now, let's keep the the comments minimal, merely referring to the equivalent C++ as the primary documentation, and only having extended comments when they are needed to describe differences from the C++. For example, maybe here we would have:

```
/// ImageSpec describes the data format of an image -- dimensions, layout,
/// number and meanings of image channels. It is equivalent to the C++
/// Imagespec, but it is only exposed to C as a pointer to an opaque structure.
typedef struct ImageSpec ImageSpec;

/// Create a new ImageSpec. The caller takes ownership and is responsible
/// for eventually calling ImageSpec_delete.
///
/// Equivalent C++:  `new Imagespec`
///
OIIO_API ImageSpec*
ImageSpec_new();

/// Delete an ImageSpec.
///
/// Equivalent C++: `delete is;`
///
OIIO_API void
ImageSpec_delete(const ImageSpec* is);

/// Create a new ImageSpec with explicit dimensions.
///
/// Equivalent C++:  `new ImageSpec(xres, yres, nchans, fmt)`
///
OIIO_API ImageSpec*
ImageSpec_new_with_dimensions(int xres, int yres, int nchans, TypeDesc fmt);

/// Width of the pixel data window. In C++ we would directly access the structure members,
/// but in C we only have a pointer to an opaque ImageSpec structure.
///
/// Equivalent C++:  `is->width`
///
OIIO_API int
ImageSpec_width(const ImageSpec* is);

```

Reply:
Yeah, that make sense.